components/cam/src/physics/cam/physics_types.F90

```diff
@@ -233,9 +235,7 @@ subroutine physics_update_main(state, ptend, dt, tend)
     integer :: ixo, ixo2, ixh, ixh2, ixn    ! indices for O, O2, H2, and N
 
     real(r8) :: zvirv(state%psetcols,pver)  ! Local zvir array pointer
-
-    real(r8),allocatable :: cpairv_loc(:,:,:)
-    real(r8),allocatable :: rairv_loc(:,:,:)
+    real(r8) :: rairv_loc(state%psetcols,pver)  ! Local zvir array pointer
```

Comment:
I also like removing runtime allocations-deallocations if not absolutely needed. These are expensive calls and sometime defragments the memory.